rust/core/src/driver_manager.rs

```diff
@@ -164,7 +164,7 @@ struct ManagedDriverInner {
 /// Implementation of [Driver].
 #[derive(Clone)]
 pub struct ManagedDriver {
-    inner: Arc<ManagedDriverInner>,
+    inner: Pin<Arc<ManagedDriverInner>>,
```

Comment:
Why do we need `Pin` here?

Reply:
I want to ensure the `FFI_AdbcDriver` doesn't get moved around after a pointer is passed to the C side. Instead of wrapping `FFI_Driver` in a `Box` (would add another pointer indirection) I pinned this `Arc` instead.

Reply:
I don't think `Pin` helps here with that? `ManagedDriverInner` implements [`Unpin`](https://doc.rust-lang.org/stable/std/marker/trait.Unpin.html).